drivers/block-nbd.c

```diff
@@ -942,7 +943,8 @@ tdnbd_open(td_driver_t* driver, const char* name,
 				ERROR("Failure to malloc for NBD destination");
 				return -1;
 			}
-			strcpy(prv->peer_ip, peer_ip);
+			strncpy(prv->peer_ip, peer_ip, strlen(peer_ip) + 1);
```

Comment:
The length here should be the allocated size of the destination, otherwise you can still overflow the destination if the source is very large.

Reply:
Dest is `malloc'd`  to that size 2 lines above and the result checked, so we wither fail to allocate or have enough space.

Reply:
`strcpy` is guaranteed to be correct here because it is allocated to the correct size so use of `strncpy` seems unnecessary here.

In any case, it would be preferable to replace the malloc+cpy with `strdup()`.